src/react/hooks/__tests__/useSubscription.test.tsx

```diff
@@ -1122,6 +1122,276 @@ followed by new in-flight setup", async () => {
   });
 });
 
+describe("ignoreResults", () => {
+  const subscription = gql`
+    subscription {
+      car {
+        make
+      }
+    }
+  `;
+
+  const results = ["Audi", "BMW"].map((make) => ({
+    result: { data: { car: { make } } },
+  }));
+
+  it("should not rerender when ignoreResults is true, but will call `onData` and `onComplete`", async () => {
+    const link = new MockSubscriptionLink();
+    const client = new ApolloClient({
+      link,
+      cache: new Cache({ addTypename: false }),
+    });
+
+    const onData = jest.fn((() => {}) as SubscriptionHookOptions["onData"]);
+    const onError = jest.fn((() => {}) as SubscriptionHookOptions["onError"]);
+    const onComplete = jest.fn(
+      (() => {}) as SubscriptionHookOptions["onComplete"]
+    );
```

Comment:
```suggestion
    const onData = jest.fn();
    const onError = jest.fn();
    const onComplete = jest.fn();
```

Out of curiosity, are these casts needed? I'm able to remove this entirely and see no type errors. If you're using this to convey intent, feel free to ignore this comment.

(this suggestion applies to any place with these mocks, but I'll avoid adding this comment to the other locations)

Reply:
It's for test-writing DX. If you hover `onData` further down in the test to get a feeling what you should assert on, you either get `any`, or the right type.